core/src/Microsoft.Teams.BotApps/Schema/TeamsConversation.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Bot.Core.Schema;
+
+namespace Microsoft.Teams.BotApps.Schema;
+
+/// <summary>
+/// Teams Conversation schema.
+/// </summary>
+public class TeamsConversation : Conversation
+{
+    /// <summary>
+    /// Initializes a new instance of the TeamsConversation class.
+    /// </summary>
+    [JsonConstructor]
+    public TeamsConversation()
+    {
+        Id = string.Empty;
+    }
+
+    /// <summary>
+    /// Creates a new instance of the TeamsConversation class from the specified Conversation object.
+    /// </summary>
+    /// <param name="conversation"></param>
+    public TeamsConversation(Conversation conversation)
+    {
+        ArgumentNullException.ThrowIfNull(conversation);
+        Id = conversation.Id ?? string.Empty;
+        if (conversation.Properties == null)
+        {
+            return;
+        }
+        if (conversation.Properties.TryGetValue("tenantId", out object? tenantObj) && tenantObj is JsonElement je && je.ValueKind == JsonValueKind.String)
+        {
+            TenantId = je.GetString();
+        }
+        if (conversation.Properties.TryGetValue("conversationType", out object? convTypeObj) && convTypeObj is JsonElement je2 && je2.ValueKind == JsonValueKind.String)
+        {
+            ConversationType = je2.GetString();
+        }
+    }
+
+    /// <summary>
+    /// Tenant Id.
+    /// </summary>
+    [JsonPropertyName("tenantId")] public string? TenantId { get; set; }
+
+    /// <summary>
+    /// Conversation Type.
+    /// </summary>
+    [JsonPropertyName("conversationType")] public string? ConversationType { get; set; }
```

Comment:
Start the static class for ConversationType (And comment here that points to that class)